### PR TITLE
Refine CMake scripts and fix g++ 5 build

### PR DESCRIPTION
**Pull request by Aesylwinn**
Should there be the option to choose between Lua or LuaJit? It should be feasible, and could be added as a choice in CMake. According to its README, Sol2 supports it.

### CODE REVIEW

File: components/openmw-mp/Base/BasePacketProcessor.hpp [hunk 1]
```diff
@@ -38,7 +39,7 @@ class BasePacketProcessor
                 throw std::logic_error("processor " + p.second->strPacketID + " already registered. Check " +
                                        processor->className + " and " + p.second->className);
         }
-        processors.insert(typename processors_t::value_type(processor->GetPacketID(), std::move(processor)));
+        processors.emplace(processor->GetPacketID(), std::unique_ptr<Proccessor>(processor));
```
Comment (Capostrophic) [inline components/openmw-mp/Base/BasePacketProcessor.hpp:42]: "Proccessor" looks wrong. Or is there a typo somewhere else in the code?

Comment (Aesylwinn) [reply components/openmw-mp/Base/BasePacketProcessor.hpp:42]: Yes, that's how the template parameter is spelled.

> `template<typename Proccessor>`

**Comment by davidcernat** [comment]
Thank you. These refinements are much appreciated.

**Comment by Aesylwinn** [comment]
Looks like I made a small mistake. I did not realize that Sol2 had been added as a git submodule. A temporary workaround is to manually set the Sol2_INCLUDE_DIR in cmake. I have added a commit in #368 to automatically search the extern directory so life will be easier.

**Pull request merged**